### ISSUE TITLE
Infinite page scrolling for:

### ISSUE THREAD

**Issue by nazar** [mikelax/forgingadventures#40]
1. Games list
2. Game messages

Ideally the same infinite page scroll component should be used for the above and any subsequent views that require it.

**Comment by nazar** [comment]
Interesting article: https://medium.com/@gethylgeorge/infinite-scrolling-in-react-using-apollo-and-react-virtualized-graphql-cursor-pagination-bf80617a8a1a

**Comment by mikelax** [comment]
I like that article, it seems to reference the cursor response structure similar to what FB Graph API does for indicating if there are more results to be returned for a given API call.

**Comment by nazar** [comment]
The apollo client docs also have an example of cursor-based scrolling as well here: https://www.apollographql.com/docs/react/recipes/pagination.html#cursor-pages

I looked at several infinite scroll component but all their issues weren't inspiring. They either had issues with react 16, cross browser issues, performance issues, reverse paging issues or were pretty large (i.e. virtualised lists) for such simple functionality.

I did find an interesting one here: https://github.com/thebuilder/react-intersection-observer

It seems to use an almost widely used Browser API to detect intersection cheaply. The only browser that isn't currently natively supported is Safari, but I've updated the [webpack polyphills to load for unsupported browsers](https://github.com/thebuilder/react-intersection-observer#polyfill-for-intersection-observer). Currently testing that component (along with cursor paging 😅 )
